core/lib/tasks/core.rake

```diff
@@ -157,4 +157,14 @@ use rake db:load_file[/absolute/path/to/sample/filename.rb]}
       puts 'DONE !   Your database should no longer have line items that are associated with deleted variants.'
     end
   end
+
+  desc 'Migrates taxon icons to spree assets after upgrading to Spree 3.4: only needed if you used taxons icons.'
+  task migrate_taxon_icons: :environment do |_t, _args|
+    Spree::Taxon.all.each do |taxon|
```

Comment:
@bbonislawski

1. please use `find_each` as some stores has a large collection of taxons
2. please filter those taxons to records that actually have an icon uploaded